src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -22,121 +22,83 @@ sealed class Http2Stream : IDisposable
             private readonly int _streamId;
             private readonly object _syncObject;
 
+            private readonly TaskCompletionSource<bool> _responseHeadersAvailable;
```

Comment:
Do we have an open issue around improving perf/allocations related to this TCS usage?  We should be able to switch now to using `ValueTask<T>` and `ManualResetValueTaskSourceCore<T>`.  We could also do that as part of a general perf push on the HTTP/2 support, which I assume we'll do once we're closer to being functionally complete.